### ISSUE TITLE
New user experience is poor

### ISSUE THREAD

**Issue by Daomephsta** [CraftTweaker/CraftTweaker-Documentation#247]
The [Getting Started](https://docs.blamejared.com/1.14/en/#getting_started/) page explains how to create a ZenScript file in excellent detail. Then it just gives the reader the code for Hello World, and explains none of it beyond where to find the output.  
Imports and variables are mentioned by name, but not explained. Instead further information on them is hidden in the Advanced Functions category.  

The page (perhaps there should be an entire category) should cover or link to documentation for at least the following:
* Creating a new ZenScript file
* Hello World in ZenScript
* Imports
* Methods
  * What they are
  * How to call them 
* Comments
* Fundamental data types. These are everything on [this page](https://docs.blamejared.com/1.12/en/#Vanilla/Variable_Types/Variable_Types/).
* Some practical things users might do with ZenScript. Linking to the docs on recipe modification would be good, it's simple and practical.
* Debugging
  * Where the log file is
  * Common errors, and how to fix them
  * That the parser may give the wrong location for a syntax error if there are many syntax errors
  * Maybe how to make a good help request? What files to include, etc.

**Comment by InQuognito** [comment]
I am currently working on rewriting a lot of the docs to make it more user friendly, especially for new or otherwise inexperienced users. Please be patient with the devs, and other contributors working to change this.

**Comment by ZariusCane** [comment]
I agree. I used crafttweaker a whole lot in 1.12.  I noticed the mod was released for 1.15.2 and no documentation is listed outside of 1.12 and 1.14.  Did nothing change from 1.14?  Do we use 1.14 documentation for 1.15.2?

**Comment by deathbutton1** [comment]
Another thing that I can't find in the documentation that I would really want to know is: how do I test if a variable can be casted to a type? Like let's say I want to do something whenever an EntityMob is hurt, but if I add a function to the onEntityHurt event it will get called for any entity that is hurt, so I need some way to test if the entity is an EntityMob.

**Comment by Daomephsta** [comment]
This is still a huge issue.

**Comment by InQuognito** [comment]
Yeah, I kinda gave up on the docs, the only way to fix it was to rebuild everything from the group up which was what I was trying to do but the devs never pushed it so I gave up.

**Comment by jaredlll08** [comment]
Yea... 
I have negative free time right now, gotta prioritize things unfortunately.


> the devs never pushed it

@justinrusso who isn't a dev, but has also helped out in the docs gave you feedback on the PR in June. 

I asked the status in August. 
https://github.com/CraftTweaker/CraftTweaker-Documentation/pull/263#issuecomment-667574167

I'm not aur ewhat more we were meant to push, all I saw was a PR that had unaddressed feedback and no replies to me asking for the status.



As for the docs in general, I don't mind rebuilding from the ground up, I know they aren't the best right now, but like I said, I really do have negative free time, so the only free time I get I gotta prioritize on other things.

**Comment by InQuognito** [comment]
Yeah I agree, I planned on addressing the feedback but life got in the way. If I get free time this summer I may pick it back up again but I think i’m going to wait until CT gets updated to 1.16 (unless it already is)

**Comment by jaredlll08** [comment]
CraftTweaker is already on 1.16

**Comment by InQuognito** [comment]
well then that answers that question. yeah i might dedicate some time to it over the break.

**Comment by InQuognito** [comment]
jaredlll08 , how would you feel about making a script library similar to the way coding language websites document their functions?

**Comment by jaredlll08** [comment]
Do you have an example of what you mean?

**Comment by InQuognito** [comment]
> Do you have an example of what you mean? 

![image](https://user-images.githubusercontent.com/44120299/100274963-e3140400-2f24-11eb-9a22-7cc670743ed7.png)

**Comment by jaredlll08** [comment]
Hey, sorry it took me long to get back to this.

I would be open to that, @kindlich do you think we could get the auto docs doing that?
Like generate a method list at the top, and have it link to the methods down below
